set_tags/utils.py

```diff
@@ -185,3 +192,49 @@ def get_access_approved_role_tag(tags):
       return access_approved_role_tag
   else:
     raise ValueError(f'Expected to find {PRINCIPAL_ARN_TAG_KEY} in {tags}')
+
+def get_marketplace_customer_id(synapse_id):
+  '''Get the Service Catalog customer ID.
+  :param synapse_id: synapse user id
+  :return the customer ID of the service catalog subscriber, None if cannot find customer ID
+  '''
+  marketplace_table_name = get_env_var_value('MARKETPLACE_ID_DYNAMO_TABLE_NAME')
+  customer_id = None
+  client = get_dynamo_client()
+  try:
+    response = client.get_item(
+      Key={
+        'userId': {
+          'S': synapse_id,
+        }
+      },
+      TableName=marketplace_table_name,
+      ConsistentRead=True,
+      AttributesToGet=[
+        'marketplaceCustomerId'
+      ]
+    )
+    customer_id = response["Item"]["marketplaceCustomerId"]["S"]
```

Comment:
Not every user will have a marketplace customer id.  There are users we let into SC by virtue of being in an approved team only. So I believe you would change this code to check for a null/missing customer id.

Reply:
I think that is handled by catching the exception below?  this method returns None if it cannot find a customer ID for any reason.

Reply:
I would avoid using exception catching to handle such a non-error case because the 'log' command on line 220 will log an error every time a non-registered user provisions an instance.

Also, would you please add tests for get_marketplace_customer_id (for both the case in which a customer is registered and that in which they are unregistered).

Reply:
refactored the method and added tests.